agenthub/thinkact_agent/plans.md

```diff
@@ -0,0 +1,36 @@
+# Things that need to be done for SWE-Agent
+
+## 1. Add in all of the following commands to ACI
+```
+'search_for': {
+    'params': '<keywords>',
+    'description': 'Will allow you to search your working directory for files and folders that match your <keyword>.',
+}
+
+'edit ': {
+    'params': '<filename>',
+    'description': 'This will allow you to modify files within your working directory. Usage: "edit example.txt" -> opens the example.txt file and shows the first 100 lines',
+},
+
+'goto': {
+    'params': '<line_num>',
+    'description': 'This will allow you to go through a file to any line. Usage: "Goto 124" -> returns lines 124-224 within current file',
+},
+
+'scroll_up': {
+    'params': '',
+    'description': 'When you are in a file you can see the 100 lines above your current view. Usage: "scroll_up" -> returns the 100 lines above what you were reading',
+},
+
+'scroll_down': {
+    'params': '',
+    'description': 'When you are in a file you can see the 100 lines below your current view. Usage: "scroll_down" -> returns the 100 lines below what you were reading',
+},
```

Comment:
We should translate our current `read` command here, e.g. adding `start_line` and `end_line` parameters

`scroll_up` and `scroll_down` seem superfluous--the agent could just `goto` cur+100. Curious why SWE-agent did this. IMO we should handle the `scroll` actions inside the Agent python rather than abstracting them out into new Actions

Reply:
I was watching a deep dive with one of the devs and he was saying that the scroll ability helps a lot when the model does not know/ recall the line it needs to go to. for example, they showed it searching for a function after it found that function in a file by scrolling. I do agree that it can probably be better achieved with 'goto' just thought I would add it to stay in line with the original swe-agent.

Reply:
A few arbitrary things I've been reading lately seemed to imply that the models often have issues with specific lines numbers/etc; so I suspect they opted for this method for a similar reason.

Reply:
Yeah agree we should stick closely to their implementation--they're doing something right!

Reply:
I don't see any implementation of the `scroll` commands. I would expect our `read` command to be modified to accept a position and number of lines (or startline/endline)